classes/core/SettingsBuilder.php

```diff
@@ -75,6 +76,42 @@ public function update(array $values)
         $us = $this->model->getSettingsTable();
         $primaryKey = $this->model->getKeyName();
 
+        // check if new settings value or multilingual value added and store those to DB
+        $modelWithGivenSettings = static::find($this->model->getKey(), $settingValues->keys()->toArray());
+        $rows = [];
+        $settingValues->each(function (mixed $settingValue, string $settingName) use ($modelWithGivenSettings, &$rows) {
+            $settingName = Str::camel($settingName);
+            
+            if (!$modelWithGivenSettings->{$settingName} || is_array($modelWithGivenSettings->{$settingName})) {
+                if ($this->isMultilingual($settingName)) {
+                    $existingLocales = array_keys($modelWithGivenSettings->{$settingName} ?? []);
+                    
+                    foreach ($settingValue as $locale => $localizedValue) {
+                        if (in_array($locale, $existingLocales)) {
+                            continue;
+                        }
+
+                        $rows[] = [
+                            $this->model->getKeyName() => $this->model->getKey(),
+                            'locale' => $locale,
+                            'setting_name' => $settingName,
+                            'setting_value' => $localizedValue,
+                        ];
+                    }
+                } else {
+                    $rows[] = [
+                        $this->model->getKeyName() => $this->model->getKey(),
+                        'locale' => '',
+                        'setting_name' => $settingName,
+                        'setting_value' => $settingValue,
+                    ];
+                }
+            }
+        });
+
+        if (count($rows) > 0) {
+            DB::table($this->model->getSettingsTable())->insert($rows);
+        }
```

Comment:
@Vitaliy-1 this particular change is seems required as when updating a model instance with new settings/multilingual column, it does not add the data in the settings table . for example , a announcement model data may not have `descriptionShort` or may have `descriptionShort` for only `en` . But if we try to update it as following 
```php
$announcement->update([
    "descriptionShort" => [
        "en" => "<p>Test Announcement 2025 nothing</p>"
        "fr_CA" => "<p>something new</p>"
    ],
]);
```
it will not add a new row in the settings table for `fr_CA` which is expected . I think this will cause issue when updating a multilingual or normal setting field as it will never get inserted . Please take a look of above implementation .

Reply:
Good spot! It seems that my idea to update settings in 1 query doesn't work if the value wasn't set. Considering the example query:
```sql
UPDATE announcement_settings SET setting_value = 
CASE WHEN setting_name='title' AND locale='en' THEN 'English Title' WHEN setting_name='title' AND locale='fr_CA' THEN 'French TItle' ELSE setting_value END
WHERE announcement_id = 16;
```
I really don't now SQL well enough to say if we can modify it to create a new row if it doesn't exist. 
On the other hand, we can just run a series of conventional sql queries to update existing rows or insert new ones. This is a quick example: https://github.com/pkp/pkp-lib/pull/10542/files
Let me know if you want me to resolve this bug

Reply:
@Vitaliy-1 I am not sure also how to update and insert in a single query and the query may become too complex to work in future too . However I think we need to consider 3 cases here 
1. Updating the existing ones in table in passed through update request .
2. Adding new ones that do not exists in table passed through update request
3. deleting what is not passed in the update request . 

for for example , if an `announcement` model instance have 4 settings row for locales `en`, `fr_CA`, `fr_FR`, `pt_BR`, for `title` prop and we make a request like 
```php
$announcement->update([
    "title" => [
        "en" => "Title in EN"
        "fr_CA" => "Title in fr_CA",
        "fr_FR" => "Title in fr_FR",
        "el" => "Title in el",
    ],
]);
```
So in the above example , we updating `en`, `fr_CA`, `fr_FR` , adding new `el` and no details about `pt_BR` which need to be removed . I have added an implementation at https://github.com/pkp/pkp-lib/pull/10324/files#diff-4dcc09fe73d537128c17c4bbb260ad4ba24ccc0ecb8cdc7d0b47ebc7986e2cedR79-R129 but that run 3 queries(update, insert and delete) . 

Maybe we can optimize in 2 queries by deleting all for that given props and inserting again. What you think ?

Reply:
In general, on the object level, all looks OK. All three conditions are met. So in theory we should have all needed info to update data once per row. Looking into it

Reply:
Aha, we have `PKPSchemaService::addMissingMultilingualValues()` for API calls that ensures consistency. And we don't have the same logic for Models.